orchagent/portsorch.cpp

```diff
@@ -2873,3 +2873,49 @@ void PortsOrch::doTask(NotificationConsumer &consumer)
         sai_deserialize_free_port_oper_status_ntf(count, portoperstatus);
     }
 }
+
+/*
+ * sync up orchagent with libsai/ASIC for port state.
+ *
+ * Currently NotificationProducer is used by syncd to inform port state change,
+ * which means orchagent will miss the signal if it happens between orchagent shutdown and startup.
+ * Syncd doesn't know whether the signal has been lost or not.
+ * Also the source of notification event is from libsai/SDK.
+ *
+ * Latest oper status for each port is retrieved via SAI_PORT_ATTR_OPER_STATUS sai API,
+ * the hostif and db are updated accordingly.
+ */
+void PortsOrch::syncUpPortState()
+{
+    SWSS_LOG_ENTER();
+
+    for (auto &it: m_portList)
```

Comment:
By design, it does the same with PortsOrch::doTask(NotificationConsumer &consumer). Is it possible to reuse code?

For example, feed status into NotificationConsumer, and let doTask process them? Or refactor a common function.

Reply:
The way current implementation works doesn't require timer to do a cutoff as that for arp and route.  I don't see the benefit of transforming status fetched from ASIC into notification message. 

Yes, they both call updateDbPortOperStatus() and setHostIntfsOperStatus(), all other processing are different due to the difference in port state source and object scope.  

In the warm start sync up phase, a little more check is done to compare with current port oper status so updateDbPortOperStatus() and setHostIntfsOperStatus() are called only when needed.  Similar check might be desired for port state change notification, in that case common function may be good.